### DISCUSSION TITLE
Build fails with error "Value of optional type 'URL?' must be unwrapped to a value of type 'URL'"

### DISCUSSION THREAD

**Discussion by yc3yu** [liamnichols/xcstrings-tool#44] [Q&A] [answered]
I followed the steps from [_Integrating XCStrings Tool into an Xcode Project Target_](https://swiftpackageindex.com/liamnichols/xcstrings-tool/0.1.2/documentation/documentation/integrating-xcstrings-tool-into-an-xcode-project-target#Configuring-Build-Settings) to set this up in my project, and I'm getting the errors shown in the screenshot below when I try to build the project.

The only difference in my integration process is that I typed a slightly different link in when adding the package dependency—the instructions say to use `github.com/liamnichols/xcstrings-tool-plugin`, but this link didn't show any results, and the link that worked for me was `https://github.com/liamnichols/xcstrings-tool.git`.

Has this happened to anyone else, and do you have a fix for it? Did I miss a step in the integration?

<img width="1440" alt="image" src="https://github.com/liamnichols/xcstrings-tool/assets/68297404/1f1e3a05-fc6b-43db-8036-1d9ba3dee7be">

**Comment by liamnichols** [comment]
Hi thanks for raising this! Which version of Xcode are you using? 

> The only difference in my integration process is that I typed a slightly different link in when adding the package dependency

It's recommended to use the xcstrings-tool-plugin because the build-tool plugin defined in this package uses a binary artefact dependency of the xcstrings-tool command line utility instead of requiring it to be built from source as part of your dependency graph.

Doing so helps to eliminate slow compile times (SwiftSyntax) and other platform incompatibilities... The issue you are seeing seems to be that the xcstrings-tool Command Line utility cannot be compiled in your environment (maybe it's a change in Xcode 15.3, I haven't used it yet...)

If you can setup your project using xcstrings-tool-plugin, this will avoid the issue you are seeing, otherwise, we might need to figure out why `output` and `input` are now optionals 🤔

**Reply by liamnichols** [reply, marked as answer]
Just a note, but I fixed the compile time issue that you see on the `main` branch (#45). It was due to a change in the latest version of swift-argument-parser.

If you want to continue using the plugin from source, you can use the `main` branch until the next release, or just pin `swift-argument-parser` back to version 1.2.3 by adding it as a fixed dependency in your Package.swift.

Thanks!

**Reply by yc3yu** [reply]
Thanks for the detailed response!

I'm using Xcode 15.3. No package results were showing up when I searched for

`github.com/liamnichols/xcstrings-tool-plugin`

as defined in the instructions article, but the other link I used (not the plugin version) returned a search result, so I used that one.

I was finally able to add xcstrings-tool-plugin as a package with a simple change, though: adding `https://` to the beginning of the link (i.e. using

`https://github.com/liamnichols/xcstrings-tool-plugin`

to search for the package). So far, this seems to be working.

Thanks for your help!
